app/scripts/views/sign_up.js

```diff
@@ -140,7 +143,10 @@ define(function (require, exports, module) {
     },
 
     showValidationErrorsEnd () {
-      if (this._isEmailSameAsBouncedEmail()) {
+      if (! this._doPasswordsMatch()) {
+        this.showValidationError('#password',
```

Comment:
I like how you tied this error to the first password field. I'll update the email-first PR to do the same.